amethyst_utils/src/autosave.rs

```diff
@@ -0,0 +1,95 @@
+//! Module containing the `AutoSaveSystem`.
+
+use amethyst_core::specs::{Resources, System, SystemData, Write};
+// It is really dirty
+use dirty::Dirty;
+use serde::{de::DeserializeOwned, Serialize};
+use std::{
+    fs::File,
+    io::{Read, Write as WriteIo},
+    marker::PhantomData,
+};
+
+/// System used to automatically save a Resource T to a file.
+/// On load, it will attempt to load it from the file and if it fails, it will use T::default().
+/// The resource in question will be wrapped into a `Dirty<T>` value inside of specs to keep track of changes made to the resource.
+/// This `System` will save the resource each time there is a modification.
```

Comment:
That's hardly a good strategy. I'm not sure if we should put that into Amethyst. The proper way would be a simple event / flag that indicates the resource should be saved. That would also remove the need to wrap it in `Dirty`.

Reply:
The other way is to have a trait on the resource, but I figured it wouldn't be usable for things already implemented like DisplayConfig

Reply:
I think having file per resource is not really good idea as it would either force user to have big resources or accept many files